extensions/resource-deployment/src/services/notebookService.ts

```diff
@@ -79,7 +79,9 @@ export class NotebookService implements INotebookService {
 	 * @param content the notebook content
 	 */
 	async launchNotebookWithContent(title: string, content: string): Promise<azdata.nb.NotebookEditor> {
-		const uri: vscode.Uri = vscode.Uri.parse(`untitled:${title}`);
+		const ext = path.extname(title) ?? 'ipynb';
+		title = path.basename(title, ext);
+		const uri: vscode.Uri = vscode.Uri.parse(`untitled:${title}-${getDateTimeString()}.${ext}`);
```

Comment:
maybe you can reuse this logic to make sure the notebook is unique: https://github.com/microsoft/azuredatastudio/blob/release/1.18/extensions/resource-deployment/src/services/notebookService.ts#L188

Reply:
Yes.